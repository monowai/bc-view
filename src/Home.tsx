import React from "react";
import { useKeycloak } from "@react-keycloak/ssr";
import { Redirect } from "react-router";

const Home = (): JSX.Element => {
  const { keycloak } = useKeycloak();
  if (keycloak?.authenticated) {
    return <Redirect to="/portfolios" />;
  }
  return <Redirect to="/login" />;
};

export default Home;
