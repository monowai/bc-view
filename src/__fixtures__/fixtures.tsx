import { UserProfile, UserProvider } from "@auth0/nextjs-auth0/client"
import React, { ReactNode } from "react"
import { PortfolioResponses, RegistrationResponse } from "types/beancounter"

export const mockUserProfile: UserProfile = {
  email: "foo@example.com",
  email_verified: true,
  name: "foo",
  nickname: "foo",
  picture: "foo.jpg",
  sub: "1",
  updated_at: null,
}
export const registrationSuccess: RegistrationResponse = {
  data: {
    id: "ownerId",
    email: "mike@bc.com",
    active: true,
    since: "2020-03-03",
  },
}

export const portfolioResult: PortfolioResponses = {
  data: [
    {
      id: "P123",
      code: "P123",
      name: "Portfolio 1",
      currency: {
        code: "EUR",
        name: "Euro",
        symbol: "â‚¬",
      },
      base: {
        code: "USD",
        name: "Dollar",
        symbol: "$",
      },
      owner: {
        id: "ownerId",
        email: "mike@bc.com",
        active: true,
        since: "2020-03-03",
      },
      marketValue: 0,
      irr: 0,
    },
  ],
}
export const withUserProvider = (
  user: UserProfile = mockUserProfile,
): React.ComponentType<{ children: ReactNode }> => {
  const TestUserProvider = ({
    children,
  }: {
    children: ReactNode
  }): React.ReactElement => <UserProvider>{children}</UserProvider>
  TestUserProvider.displayName = `TestUserProvider(${user.name})`
  return TestUserProvider
}
