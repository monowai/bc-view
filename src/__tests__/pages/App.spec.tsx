import Home from "@/pages/index";
import React from "react";
import { screen, render } from "@testing-library/react";
import { mockUser, withUserProvider } from "../fixtures";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ ready: true, t: (key: string) => key }),
}));

afterEach(() => {
  jest.clearAllMocks();
  jest.restoreAllMocks();
  jest.resetModules();
});

describe("<App />", () => {
  test("renders for authorised user", () => {
    const { container } = render(<Home user={mockUser} />, {
      wrapper: withUserProvider({ user: mockUser }),
    });
    const heading = screen.getByText("home.welcome");
    expect(heading).toBeInTheDocument();
    expect(screen.getByText("home.portfolios")).toBeInTheDocument();
    expect(screen.getByText("user.logout")).toBeInTheDocument();
    expect(container).toMatchSnapshot()
  });
});
