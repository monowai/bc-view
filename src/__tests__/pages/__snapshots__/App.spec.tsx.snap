// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders for authorised user 1`] = `
<div>
  <div>
    home.welcome
    <div>
      <a
        href="/portfolios"
      >
        home.portfolios
      </a>
    </div>
    <div>
      <a
        href="/api/auth/logout"
      >
        user.logout
      </a>
    </div>
  </div>
</div>
`;
